test/foundry/unit/concrete/erc4337account/TestERC4337Account_AddDeposit.t.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "../../../utils/Imports.sol";
+import "../../../utils/SmartAccountTestLab.t.sol";
+
+    event DepositAdded(address indexed account, address indexed depositor, uint256 amount);
+
+contract TestERC4337Account_addDeposit is Test, SmartAccountTestLab {
+    SmartAccount private account;
+
+    function setUp() public {
+                init();
+                account = BOB_ACCOUNT;
+    }
+
+    function test_AddDeposit_Success() public {
+        uint balanceBefore = ENTRYPOINT.balanceOf(address(account));
+        uint256 depositAmount = 1 ether;
+
+        account.addDeposit{value: depositAmount}();
+
+        assertEq(balanceBefore + depositAmount, ENTRYPOINT.balanceOf(address(account)), "Deposit should be added to EntryPoint");
+    }
+
+    function test_AddDeposit_EventEmitted() public {
+        uint256 depositAmount = 1 ether;
+        (bool res, ) = address(account).call{value: depositAmount}(""); // Pre-funding the account contract
+        assertTrue(res, "Pre-funding account should succeed");
```

Comment:
what does it mean "pre-funding"
would this mean undeployed account counterfactual?